compiler/rustc_middle/src/ty/flags.rs

```diff
@@ -122,22 +116,12 @@ impl FlagComputation {
             }
 
             ty::CoroutineWitness(_, args) => {
-                let should_remove_further_specializable =
-                    !self.flags.contains(TypeFlags::STILL_FURTHER_SPECIALIZABLE);
                 self.add_args(args);
-                if should_remove_further_specializable {
-                    self.flags -= TypeFlags::STILL_FURTHER_SPECIALIZABLE;
-                }
             }
 
             &ty::Closure(_, args) => {
                 let args = args.as_closure();
```

Comment:
i feel like this can and should just be `self.add_args(args)` now 🤔 same for coroutine closure

Reply:
yeah, at this point it can be fully structural.